### PR TITLE
Domains: Populate the currentUser currencyCode from the productList

### PR DESCRIPTION
**Pull request by hambai**
The current user currencyCode is populated from the plans API endpoint. However in domains-only flow we don't load the plans at all since there is no need for them to load and this makes the prices shown in the domain search component to be paired with the wrong currency (if you're in India you'll see the prices in USD but the actual price will be in rupee - so it'll look like more than 1000 USD for domain). This PR should populate the currentUser.currency even when you open the domains-only flow.

Depends on D26748-code

#### Changes proposed in this Pull Request

* Populate the currentUser.currencyCode when the productList is populated

#### Testing instructions

* Apply D26748-code
* Open http://calypso.localhost:3000/start/domain/domain-only in incognito window
* In the console check that `state.currentUser.currencyCode` has value
* search for a domain and make sure the prices are in the correct currency
* Also change your user default currency
* Open http://calypso.localhost:3000/start/domain/domain-only logged in as yourself
* Verify that the prices are in the correct currency (try different currencies as INR and etc)

Fixes #

https://core.trac.wordpress.org/ticket/46756

### CODE REVIEW

File: client/state/current-user/reducer.js [hunk 1]
```diff
@@ -84,7 +92,7 @@ export const currencyCode = createReducer(
 export function capabilities( state = {}, action ) {
 	switch ( action.type ) {
 		case SITE_RECEIVE:
-		case SITES_RECEIVE:
```
Comment (aidvu) [inline client/state/current-user/reducer.js]: Why the brackets?

**Comment by hambai** [comment]
@aidvu 

> Tho we should stop doing this in both Plans and Products List. The currency should be injected into the user object right away, and actually returned from the user object, not like this.

I agree, but currently when you are not logged in there is no user object or to be precise - there is no API endpoint that populated the currentUser object. So that would require a bigger refactoring and could break a lot more things :)

File: client/state/current-user/reducer.js (hunk 1 above)
Comment (hambai) [reply client/state/current-user/reducer.js]: Linter

**Comment by aidvu** [comment]
> I agree, but currently when you are not logged in there is no user object or to be precise - there is no API endpoint that populated the currentUser object. So that would require a bigger refactoring and could break a lot more things :)

I believe we have server-side code in node.js that fills this info. It would be fairly easy to just pull the currency there.

**Pull request merged**